src/main/kotlin/calculator/CustomStringDelimiter.kt

```diff
@@ -0,0 +1,50 @@
+package calculator
+
+import kotlin.streams.toList
+
+class CustomStringDelimiter(value: String?) {
+
+    private val values: List<String>
+    private val operands: MutableList<Int>
+    private val operators: MutableList<String>
+
+    init {
+
+        values = validate(value)
+        operators = values.stream()
+            .filter { Expression.STRING_OPERATORS.contains(it) }
+            .toList()
+            .toMutableList()
+
+        operands = values.stream()
+            .filter { !Expression.STRING_OPERATORS.contains(it) }
+            .map { it.toInt() }
+            .toList()
+            .toMutableList()
+    }
+
+    private fun validate(value: String?): List<String> {
+        if (value == null) {
+            throw IllegalArgumentException("잘못된 값이 전달되었습니다.")
+        }
+
+        if (value.isBlank() || value.isEmpty()) {
+            throw IllegalArgumentException("잘못된 값이 전달되었습니다.")
+        }
+
+        return value.split(DELIMITER)
+            .toList()
+    }
+
+    fun getOperands(): MutableList<Int> {
+        return operands.toMutableList()
+    }
```

Comment:
```suggestion
    fun getOperands(): MutableList<Int> = operands
```

코틀린 Collections API를 사용하면 자바처럼 toList를 붙여주지 않아도 됩니다. 😊

또한 문(statement) 보다 식(expression)를 사용하여 간결하게 표현이 가능합니다 🤩